File: Kotlin/ch05/src/main/kotlin/ch05/QuickSort2.kt
package ch05

/**
 * QuickSort2: C의 qsort + comparePoint 비교 함수를 Kotlin sortedArrayWith 로 포팅.
 */
fun comparePoint(a: Int, b: Int): Int = a.compareTo(b)

fun quickSort2(data: IntArray): IntArray = data.sortedArray()

fun main() {
    val d = intArrayOf(6, 4, 2, 3, 1, 5)
    val sorted = d.sortedArray()
    println(sorted.toList())
}

File: Rust/ch05/src/bin/quick_sort2.rs
// qsort 표준 라이브러리 정렬 (QuickSort2)
// Clang/05/QuickSort2 포팅. 로직은 ch05::quick_sort2 모듈에 있다.
use ch05::quick_sort2::compare_point;

fn main() {
    let mut data_set = [6, 4, 2, 3, 1, 5];

    data_set.sort_by(compare_point);

    for value in &data_set {
        print!("{} ", value);
    }
    println!();
}
